pkg/provider/aws/aws.go

```diff
@@ -0,0 +1,149 @@
+		ssm: ssm.New(config.session),
+	}
+}
+
+func (p *Provider) LoadSecrets(_ context.Context, paths []string) ([]provider.Secret, error) {
+	var secrets []provider.Secret
+
+	for _, path := range paths {
+		split := strings.SplitN(path, "=", 2)
+		originalKey, secretID := split[0], split[1]
+
+		// valid secretsmanager secret examples:
+		// arn:aws:secretsmanager:region:account-id:secret:secret-name
+		// secretsmanager:secret-name
+		if strings.Contains(secretID, "secretsmanager:") {
+			secret, err := p.sm.GetSecretValue(
+				&secretsmanager.GetSecretValueInput{
+					SecretId: aws.String(secretID),
+				})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get secret from AWS secrets manager")
+			}
+
+			if json.Valid([]byte(aws.StringValue(secret.SecretString))) {
+				var secretValue map[string]interface{}
+				err := json.Unmarshal([]byte(aws.StringValue(secret.SecretString)), &secretValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to unmarshal secret value")
+				}
+
+				// If there is only one key in the secret, append it directly
+				if len(secretValue) == 1 {
+					for _, value := range secretValue {
+						secretToAppend, err := appendSecret(originalKey, value)
+						if err != nil {
+							return nil, errors.Wrap(err, "failed to append secret")
+						}
+
+						secrets = append(secrets, secretToAppend)
+					}
+
+					continue
+				}
+
+				// If the secret is a JSON object, append it as a single secret
+				JSONValue, err := json.Marshal(secretValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to marshal secret value")
+				}
+
+				secretToAppend, err := appendSecret(originalKey, JSONValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to append secret")
+				}
+
+				secrets = append(secrets, secretToAppend)
+			} else {
+				// If the secret is not a JSON object, append it directly
+				secretToAppend, err := appendSecret(originalKey, aws.StringValue(secret.SecretString))
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to append secret")
+				}
+
+				secrets = append(secrets, secretToAppend)
+			}
+		}
+
+		// Valid ssm parameter examples:
+		// arn:aws:ssm:region:account-id:parameter/path/to/parameter-name
+		// arn:aws:ssm:us-west-2:123456789012:parameter/my-parameter
+		if strings.Contains(secretID, "ssm:") {
+			parameteredSecret, err := p.ssm.GetParameter(
+				&ssm.GetParameterInput{
+					Name:           aws.String(secretID),
+					WithDecryption: aws.Bool(true),
+				})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get parameter from AWS SSM")
+			}
+
+			secretToAppend, err := appendSecret(originalKey, aws.StringValue(parameteredSecret.Parameter.Value))
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to append secret")
+			}
+
+			secrets = append(secrets, secretToAppend)
+		}
+	}
+
+	return secrets, nil
+}
+
+func appendSecret(key string, value interface{}) (provider.Secret, error) {
```

Comment:
this can be removed since we will be able to append directly where needed, but also because this obfuscates the actual append